library/core/class.session.php

```diff
@@ -56,6 +56,51 @@ public function addPermissions($perms) {
     }
 
     /**
+     * Check the given permission, but also return true if the user has a higher permission.
+     *
+     * @param mixed $permission The permission (or array of permissions) to check.
+     * @param bool $fullMatch If $Permission is an array, $FullMatch indicates if all permissions specified are required. If false, the user only needs one of the specified permissions.
+     * @param string $junctionTable The name of the junction table for a junction permission.
+     * @param in $junctionID The ID of the junction permission.
+     * * @return boolean True on valid authorization, false on failure to authorize
+     */
+    public function checkRankedPermission($permission, $fullMatch = true, $junctionTable = '', $junctionID = '') {
```

Comment:
Ditch the `$fullMatch = true, $junctionTable = '', $junctionID = ''` parameters for now. I really don't like those and would rather they stay out until we can refactor into a nicer representation.